### PR TITLE
introduce formik select component

### PR DESCRIPTION
**Pull request by benjiwheeler**
### Resolves:

Step towards resolving #3053

### Changes:

Introduce `formik-select.jsx` component, modeled on `formik-input.jsx` but handling an `options` prop

### CODE REVIEW

File: src/components/formik-forms/formik-select.jsx [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+FormikSelect.propTypes = {
+    className: PropTypes.string,
+    error: PropTypes.string,
+    // expect structure like: [
+    //    {value: 'US', label: 'United States'}
+    //    {value: 'AG', label: 'Angola'}
+    //    ...
+    // ]
+    options: PropTypes.arrayOf(PropTypes.any),
```
Comment (picklesrus) [inline src/components/formik-forms/formik-select.jsx]: If we have specific requirements, about the object shape, I think we should write a more specific validator. I tried the code below out and it seems to work as I hoped:
options: PropTypes.arrayOf(PropTypes.shape({
     value: PropTypes.string.isRequired,
     label: PropTypes.string.isRequired,
 })),

File: src/components/formik-forms/formik-select.jsx [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+    validationClassName: PropTypes.string,
+    value: PropTypes.oneOfType([PropTypes.number, PropTypes.string])
```
Comment (picklesrus) [inline src/components/formik-forms/formik-select.jsx]: Does the number type work? I tried passing 1 and 2 to my list of length 2 and it didn't change what was selected. Perhaps I'm using it wrong though? passing Using your example in comments above "AG" worked fine though.

Comment (benjiwheeler) [reply src/components/formik-forms/formik-select.jsx]: You're right, I never added the code to set the selected value!

The number type shouldn't be the problem, though.

File: src/components/formik-forms/formik-select.jsx (hunk 1 above)
Comment (benjiwheeler) [reply src/components/formik-forms/formik-select.jsx]: Good idea, thanks!

File: src/components/formik-forms/formik-select.jsx [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+        </div>
+    );
+};
+
+
+FormikSelect.propTypes = {
+    className: PropTypes.string,
+    error: PropTypes.string,
+    // expect structure like: [
+    //    {value: 'US', label: 'United States'}
```
[the remark this answers is not in the document]
Comment (benjiwheeler) [reply src/components/formik-forms/formik-select.jsx]: Done

File: src/components/formik-forms/formik-select.jsx [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+    const optionsList = options.map((item, index) => (
+        <option
+            key={index}
+            value={item.value}
+        >
+            {item.label}
+        </option>
+    ));
+    return (
+        <div className="col-sm-9 row row-relative">
```
[the remark this answers is not in the document]
Comment (benjiwheeler) [reply src/components/formik-forms/formik-select.jsx]: You're right, removing

**Pull request merged**